src/Metadata/Property/Factory/XmlPropertyMetadataFactory.php

```diff
@@ -0,0 +1,177 @@
+    }
+
+    /**
+     * Extracts metadata from the XML tree.
+     *
+     * @param string $resourceClass
+     * @param string $propertyName
+     *
+     * @throws InvalidArgumentException
+     *
+     * @return array
+     */
+    private function getMetadata(string $resourceClass, string $propertyName) : array
+    {
+        foreach ($this->paths as $path) {
+            try {
+                $domDocument = XmlUtils::loadFile($path, self::RESOURCE_SCHEMA);
+            } catch (\InvalidArgumentException $e) {
+                throw new InvalidArgumentException($e->getMessage(), $e->getCode(), $e);
+            }
+
+            $properties = (new \DOMXPath($domDocument))->query(sprintf('//resources/resource[@class="%s"]/property[@name="%s"]', $resourceClass, $propertyName));
+
+            if (false === $properties || 0 >= $properties->length || null === $properties->item(0) || false === $property = simplexml_import_dom($properties->item(0))) {
+                continue;
+            }
+
+            return [
+                'description' => (string) $property['description'] ?: null,
+                'readable' => 'true' === (string) $property['readable'] || 'false' === (string) $property['readable'] ? constant((string) $property['readable']) : null,
+                'writable' => 'true' === (string) $property['writable'] || 'false' === (string) $property['writable'] ? constant((string) $property['writable']) : null,
+                'readableLink' => 'true' === (string) $property['readableLink'] || 'false' === (string) $property['readableLink'] ? constant((string) $property['readableLink']) : null,
+                'writableLink' => 'true' === (string) $property['writableLink'] || 'false' === (string) $property['writableLink'] ? constant((string) $property['writableLink']) : null,
+                'required' => 'true' === (string) $property['required'] || 'false' === (string) $property['required'] ? constant((string) $property['required']) : null,
+                'identifier' => 'true' === (string) $property['identifier'] || 'false' === (string) $property['identifier'] ? constant((string) $property['identifier']) : null,
+                'iri' => (string) $property['iri'] ?: null,
+                'attributes' => $this->getAttributes($property),
+            ];
+        }
+
+        return [];
+    }
+
+    /**
+     * Recursively transforms an attribute structure into an associative array.
+     *
+     * @param \SimpleXMLElement $element
+     *
+     * @return array
+     */
+    private function getAttributes(\SimpleXMLElement $element) : array
+    {
+        $attributes = [];
+        foreach ($element->attribute as $attribute) {
+            $value = isset($attribute->attribute[0]) ? $this->getAttributes($attribute) : (string) $attribute;
+            isset($attribute['name']) ? $attributes[(string) $attribute['name']] = $value : $attributes[] = $value;
```

Comment:
I find this hard to read, I'm not found of ternary value attributions, maybe it's just me